### ISSUE TITLE
DB Accessing Pages Very Slow

### ISSUE THREAD

**Issue by safwansamsudeen** [safwansamsudeen/infilib#1]
## Issue
In v0.1, Svelte `load` functions make a lot of DB calls - all the members/items/transactions, and details about those transactions (for adding).

Furthermore, `columns`, which is far better off as a constant, is currently a function that's utilized very frequently.

In the local dev environment, there isn't any noticeable lag. In production, however, members takes about 3 s and items about double that. Worse, as SvelteKit is SSR or MPA or whatever, there's no visual update to let the user know that the a new page is being loaded.

## Potential Enhancement
Ideally, pages would respond _immediately_. For this `columns` has to be a constant - which would be fine if the select options weren't linked to the DB.

Ideally, select options could take in some values and get DB data asynchronously. The same applies to table data.

## Code suggestions
I vaguely remember reading about this in Svelte docs, and I don't think it'll be too tough. If there's no direct way to do this, we must move to a more API based version.

**Comment by safwansamsudeen** [comment]
And indeed there is a fairly easy way - we must [utilize nested promises](https://kit.svelte.dev/docs/load#streaming-with-promises).

This is my code for items now:
```js
return {
	columns: itemColumns,
	inputColumns: others,
	promises: {
		items: new Promise(async (fulfil) => {
			let items = await item.findMany({
				include,
				where: params
			});
			type && flatten(items, type);
			standardizeSelects(items, itemColumns);
			fulfil(items);
		})
	}
};
```
Somewhat unwieldy, but I'm trying to pretty it. This now works.

**Comment by safwansamsudeen** [comment]
Yesterday, on the 5th of December, I _finally_ figured out what this was all about: logging in.

Specifically, `jwtVerify`'s call to the Passage API wasn't caching properly, I guess 😅.

Exhausting, but will finally be solved in v0.4
